core/src/main/java/org/jruby/truffle/nodes/control/DoWhileNode.java

```diff
@@ -67,6 +65,7 @@ public boolean executeRepeating(VirtualFrame frame) {
                     break;
                 } catch (RedoException e) {
                     // Just continue in the while(true) loop.
+                    context.getSafepointManager().poll();
```

Comment:
But the RedoException branch is only taken when using the `redo`, keyword, no?
And that is quite rare, so there is no more safepoint in a typical loop with this code if I see right (or is it done with instrumentation by the LoopNode?).